coding/json/stream-go1.4.go

```diff
@@ -0,0 +1,206 @@
+// +build !go1.5
+
+package json
+
+import (
+	"github.com/ipfs/go-ipld/Godeps/_workspace/src/github.com/golang/go/src/encoding/json"
+	"fmt"
+	"io"
+	"sync"
+
+	reader "github.com/ipfs/go-ipld/stream"
+	mc "github.com/jbenet/go-multicodec"
+)
+
+const HeaderPath string = "/json"
+const HeaderMsgioPath string = "/json/msgio"
+
+var Header []byte
+var HeaderMsgio []byte
+var ErrAlreadyRead error = fmt.Errorf("Stream already read: unable to read it a second time")
+
+func init() {
+	Header = mc.Header([]byte(HeaderPath))
+	HeaderMsgio = mc.Header([]byte(HeaderMsgioPath))
+}
+
+type JSONDecoder struct {
+	r    io.ReadSeeker
+	pos  int64
+	lock sync.Mutex
+}
+
+type jsonParser struct {
+	reader.BaseReader
+	decoder *json.Decoder
+}
+
+func NewJSONDecoder(r io.ReadSeeker) (*JSONDecoder, error) {
+	offset, err := r.Seek(0, 1)
+	if err != nil {
+		return nil, err
+	}
+	return &JSONDecoder{r, offset, sync.Mutex{}}, nil
+}
+
+func (d *JSONDecoder) Read(cb reader.ReadFun) error {
+	d.lock.Lock()
+	defer d.lock.Unlock()
+
+	if d.pos == -2 {
+		return ErrAlreadyRead
+	} else if d.pos == -1 {
+		d.pos = -2
+	} else {
+		newoffset, err := d.r.Seek(d.pos, 0)
+		if err != nil {
+			return err
+		} else if newoffset != d.pos {
+			return fmt.Errorf("Failed to seek to position %d", d.pos)
+		}
+	}
+	jsonParser := &jsonParser{
+		reader.CreateBaseReader(cb),
+		json.NewDecoder(d.r),
+	}
+	err := jsonParser.readValue()
+	if err == reader.NodeReadAbort {
+		err = nil
+	}
+	return err
+}
+
+func (p *jsonParser) readValue() error {
+	token, err := p.decoder.Token()
+	if err != nil {
+		return err
+	}
+	//log.Printf("JSON: read token value %#v %T", token, token)
+	if delim, ok := token.(json.Delim); ok {
```

Comment:
yikes, no. this package should not have any json parsing. if no stream json parser exists that has hooks in it to add our callbacks, either this should become one, or just forget the stream parsing. 

stream parsing json is already a bad idea and if people are doing it they're going to have a bad time anyway.